modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java

```diff
@@ -243,6 +243,27 @@ public Sku postProductIdSku(Long id, Sku sku) throws Exception {
 		return _addOrUpdateSKU(cpDefinition, sku);
 	}
 
+	@Override
+	public Sku updateSkuExternalReferenceCodeToNewExternalReferenceCode(
+			String externalReferenceCode, String newExternalReferenceCode)
+		throws Exception {
+
+		CPInstance cpInstance = _cpInstanceService.fetchByExternalReferenceCode(
+			externalReferenceCode, contextCompany.getCompanyId());
+
+		if (cpInstance == null) {
+			throw new NoSuchCPInstanceException(
+				"Unable to find SKU with external reference code " +
+					externalReferenceCode);
+		}
+
+		Sku sku = _toSku(cpInstance.getCPInstanceId());
+
+		sku.setExternalReferenceCode(newExternalReferenceCode);
+
+		return _updateSKU(cpInstance, sku);
```

Comment:
```suggestion
		_cpInstanceService.updateExternalReferenceCode(newExternalReferenceCode, cpInstance.getCPInstanceId());
return _toSku(cpInstance.getCPInstanceId());
```

Reply:
this method is to add and it is just like the public CPDefinition updateExternalReferenceCode(
			String externalReferenceCode, long cpDefinitionId)
		throws PortalException;
 from CPDefinitionService.java